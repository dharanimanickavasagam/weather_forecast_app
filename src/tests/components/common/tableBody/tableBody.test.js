import React from "react";
import { shallow } from "enzyme";
import TableBody from "../../../../components/common/tableBody";

let mockData, wrapper;

beforeEach(() => {
  mockData = [{ pressure: 1017, humidity: 87, min_temp: 46, max_temp: 52 }];
  wrapper = shallow(<TableBody data={mockData} />);
});

describe("it renders component TableBody", () => {
  test("render the table data with the passed data props", () => {
    expect(wrapper).toMatchSnapshot();

    const tbody = wrapper.find("tbody");
    expect(tbody).toHaveLength(1);

    const rows = tbody.find("tr");
    expect(rows).toHaveLength(mockData.length);

    rows.forEach((tr, rowIndex) => {
      const cells = tr.find("td");
      expect(parseInt(cells.at(0).text())).toEqual(mockData[rowIndex].pressure);
      expect(parseInt(cells.at(1).text())).toEqual(mockData[rowIndex].humidity);
      expect(parseInt(cells.at(2).text())).toEqual(mockData[rowIndex].min_temp);
      expect(parseInt(cells.at(3).text())).toEqual(mockData[rowIndex].max_temp);
    });
  });
});
