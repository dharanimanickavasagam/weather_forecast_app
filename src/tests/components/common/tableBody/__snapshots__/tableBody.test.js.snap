// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders component TableBody render the table data with the passed data props 1`] = `
<tbody>
  <tr
    key="0"
  >
    <td
      key="0"
    >
      1017
    </td>
    <td
      key="1"
    >
      87
    </td>
    <td
      key="2"
    >
      46
    </td>
    <td
      key="3"
    >
      52
    </td>
  </tr>
</tbody>
`;
